sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/autofailover/ReplicaLookUp.java

```diff
@@ -0,0 +1,220 @@
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.InitialDirContext;
+
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import com.azure.spring.cloud.appconfiguration.config.implementation.AppConfigurationReplicaClientsBuilder;
+import com.azure.spring.cloud.appconfiguration.config.implementation.properties.AppConfigurationProperties;
+import com.azure.spring.cloud.appconfiguration.config.implementation.properties.ConfigStore;
+
+@Component
+public class ReplicaLookUp {
+
+    private static final String ORIGIN_PREFIX = "dns:/_origin._tcp.";
+
+    private static final String REPLICA_PREFIX_ALT = "dns:/_alt";
+
+    private static final String REPLICA_PREFIX_TCP = "._tcp.";
+
+    private static final String SRC_RECORD = "SRV";
+
+    private static final String[] TRUSTED_DOMAIN_LABELS = { "azconfig", "appconfig" };
+
+    private static final Duration FALLBACK_CLIENT_REFRESH_EXPIRED_INTERVAL = Duration.ofHours(1);
+
+    private static final Duration MINIMAL_CLIENT_REFRESH_INTERVAL = Duration.ofSeconds(30);
+
+    InitialDirContext context;
+
+    private Map<String, List<SRVRecord>> records = new HashMap<String, List<SRVRecord>>();
+
+    private Map<String, Instant> wait = new HashMap<>();
+
+    private final AppConfigurationProperties properties;
+
+    private final Semaphore semaphore;
+
+    public ReplicaLookUp(AppConfigurationProperties properties) throws NamingException {
+        this.properties = properties;
+        this.context = new InitialDirContext();
+        this.semaphore = new Semaphore(1);
+    }
+
+    @Async
+    public void updateAutoFailoverEndpoints() {
+        if (semaphore.tryAcquire()) {
+            for (ConfigStore configStore : properties.getStores()) {
+                if (!configStore.isEnabled()) {
+                    continue;
+                }
+                String mainEndpoint = configStore.getEndpoint();
+
+                List<String> providedEndpoints = new ArrayList<>();
+                if (configStore.getConnectionStrings().size() > 0) {
+                    providedEndpoints = configStore.getConnectionStrings().stream().map(connectionString -> {
+                        return (AppConfigurationReplicaClientsBuilder
+                            .getEndpointFromConnectionString(connectionString));
+                    }).toList();
+                } else if (configStore.getEndpoints().size() > 0) {
+                    providedEndpoints = configStore.getEndpoints();
+                } else {
+                    providedEndpoints = List.of(configStore.getEndpoint());
+                }
+
+                try {
+                    List<SRVRecord> srvRecords = findAutoFailoverEndpoints(mainEndpoint, providedEndpoints);
+
+                    srvRecords.sort((SRVRecord a, SRVRecord b) -> a.compareTo(b));
```

Comment:
Do we need to sort this `srvRecords`? Is there no priority order for the user-defined endpoints array in yaml file?
If we sort this array, then it doesn't follow the order that user defined.

Reply:
This came up in the design review. We plan on supporting the SRV priority and weight values. 

When a user defines multiple stores, we follow the list they provide, top down where the top option is the highest priority.